tasks/github_tasks.py

```diff
@@ -188,3 +189,60 @@ def get_token_from_app(_, app_id_env='GITHUB_APP_ID', pkey_env='GITHUB_KEY_B64')
     from .libs.common.github_api import GithubAPI
 
     GithubAPI.get_token_from_app(app_id_env, pkey_env)
+
+
+@task
+def assign_team_label(_, pr_id=-1):
+    """
+    Assigns the github team label name if a single team can
+    be deduced from the changed files
+    """
+    from .libs.common.github_api import GithubAPI
+
+    def get_team(changed_files) -> str | None:
+        codeowners = read_owners('.github/CODEOWNERS')
+
+        global_team = None
+        for file in changed_files:
+            owners = [name for (kind, name) in codeowners.of(file) if kind == 'TEAM']
+            if len(owners) != 1:
+                # Multiple / no owners case
+                return
+            else:
+                file_team = owners[0]
+                if global_team is not None and file_team != global_team:
+                    # Multiple owners case
+                    return
+                else:
+                    global_team = file_team
+
+        return global_team
+
+    gh = GithubAPI('DataDog/datadog-agent')
+
+    # Skip if necessary
+    labels = set(gh.get_pr_labels(pr_id))
```

Comment:
why do you need it to be a `set`?

Reply:
Usually I use a set when I need `in` as the time complexity is `O(1)` instead of `O(n)`

Reply:
Is it better to not use a set ?